retworkx/custom_return_types.pyi

```diff
@@ -0,0 +1,62 @@
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# This file contains only type annotations for PyO3 functions and classes
+# For implementation details, see __init__.py and lib.rs
+
+from typing import Any, Generic, List, Iterable, Mapping, TypeVar, Tuple, overload
+from collections.abc import ABC, Sequence
+
+# Until PEP 673 is implemented in Python 3.11, we need to use this hack
+Self = TypeVar("Self", bound="RetworkxCustomVecIter")
```

Comment:
I think you can get this from `typing_extensions` already if you'd rather, but that's no big deal.  Recent versions of `mypy` tend to support things as soon as they're specified in PEPs as well, I think?

Reply:
Thanks for the heads up, I switched to using `typing_extensions`